sdk/core/azure-core/src/main/java/com/azure/core/util/implementation/FileContent.java

```diff
@@ -0,0 +1,139 @@
+
+        if (chunkSize <= 0) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                    "'chunkSize' cannot be less than or equal to 0."));
+        }
+        this.file = file;
+        this.chunkSize = chunkSize;
+        if (!file.toFile().exists()) {
+            throw LOGGER.logExceptionAsError(new UncheckedIOException(
+                    new FileNotFoundException("File does not exist " + file)));
+        }
+
+        this.length = file.toFile().length();
+    }
+
+    @Override
+    public Long getLength() {
+        return this.length;
+    }
+
+    @Override
+    public String toString() {
+        return new String(toBytes(), StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public byte[] toBytes() {
+        byte[] data = this.bytes.get();
+        if (data == null) {
+            bytes.set(getBytes());
+            data = this.bytes.get();
+        }
+        return data;
+    }
+
+    @Override
+    public <T> T toObject(TypeReference<T> typeReference, ObjectSerializer serializer) {
+        return serializer.deserialize(toStream(), typeReference);
+    }
+
+    @Override
+    public InputStream toStream() {
+        try {
+            return new BufferedInputStream(new FileInputStream(file.toFile()), chunkSize);
+        } catch (FileNotFoundException e) {
+            throw LOGGER.logExceptionAsError(new UncheckedIOException("File not found " + file, e));
+        }
+    }
+
+    @Override
+    public ByteBuffer toByteBuffer() {
+        try {
+            FileChannel fileChannel = FileChannel.open(file);
+            return fileChannel.map(FileChannel.MapMode.READ_ONLY, 0, length);
+        } catch (IOException exception) {
+            throw LOGGER.logExceptionAsError(new UncheckedIOException(exception));
+        }
+    }
+
+    @Override
+    public Flux<ByteBuffer> toFluxByteBuffer() {
+        return Flux.using(() -> FileChannel.open(file), channel -> Flux.generate(() -> 0, (count, sink) -> {
+            if (count == length) {
+                sink.complete();
+                return count;
+            }
+
+            int readCount = (int) Math.min(chunkSize, length - count);
+            try {
+                sink.next(channel.map(FileChannel.MapMode.READ_ONLY, count, readCount));
+            } catch (IOException ex) {
+                sink.error(ex);
+            }
+
+            return count + readCount;
+        }), channel -> {
+            try {
+                channel.close();
+            } catch (IOException ex) {
+                throw LOGGER.logExceptionAsError(Exceptions.propagate(ex));
+            }
+        });
+    }
+
+    private byte[] getBytes() {
+
+        return FluxUtil.collectBytesInByteBufferStream(toFluxByteBuffer())
+                // this doesn't seem to be working (newBoundedElastic() didn't work either)
+                // .publishOn(Schedulers.boundedElastic())
+                .share()
+                .block();
```

Comment:
Not a fan of using `block`, could this instead be implemented with Files.readAllBytes?

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#readAllBytes-java.nio.file.Path-